### PR TITLE
Issue #98 - refactor FHIRValidator and FHIRPathEvaluator

### PR DESCRIPTION
**Pull request by JohnTimm**
Signed-off-by: John T.E. Timm <johntimm@us.ibm.com>

### CODE REVIEW

File: fhir-model/src/main/java/com/ibm/watson/health/fhir/model/util/FHIRUtil.java [hunk 1]
```diff
@@ -835,4 +836,15 @@ public static boolean isFailure(IssueSeverity severity) {
             return true;
         }
     }
+
+    public static <K, V> Map<K, V> createLRUCache(int maxEntries) {
+        return Collections.synchronizedMap(new LinkedHashMap<K, V>(maxEntries, 0.75f, true) {
```
Comment (lmsurpre) [inline fhir-model/src/main/java/com/ibm/watson/health/fhir/model/util/FHIRUtil.java]: is it worth having a dedicated class for the anonymous inner class that extends LinkedHashMap ?

Comment (JohnTimm) [reply fhir-model/src/main/java/com/ibm/watson/health/fhir/model/util/FHIRUtil.java]: Perhaps if it gets more widely used. Right now it’s only used in 2-3 places in FHIREvaluator.

File: fhir-validation/src/main/java/com/ibm/watson/health/fhir/validation/FHIRValidator.java [hunk 2]
```diff
@@ -36,44 +34,39 @@
...
         }
     }
     
-    public static FHIRValidator validator(FHIRPathTree tree) {
-        return new FHIRValidator(tree);
-    }
-    
-    public static FHIRValidator validator(Resource resource) {
-        return new FHIRValidator(FHIRPathTree.tree(resource));
+    public static FHIRValidator validator() {
```
Comment (lmsurpre) [inline fhir-validation/src/main/java/com/ibm/watson/health/fhir/validation/FHIRValidator.java]: doesn't seem to be a singleton pattern, so why not just let them new it themselves at this point?

Comment (JohnTimm) [reply fhir-validation/src/main/java/com/ibm/watson/health/fhir/validation/FHIRValidator.java]: 3 reasons:
1. I want to control construction
2. method chaining is a little cleaner looking
3. it follows the patterns used everywhere else in the code base

File: fhir-core/src/main/java/com/ibm/watson/health/fhir/core/util/LRUCache.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.watson.health.fhir.core.util;
+
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class LRUCache<K, V> extends LinkedHashMap<K, V> {
```
Comment (lmsurpre) [inline fhir-core/src/main/java/com/ibm/watson/health/fhir/core/util/LRUCache.java]: javadoc?

Comment (JohnTimm) [reply fhir-core/src/main/java/com/ibm/watson/health/fhir/core/util/LRUCache.java]: added javadoc

Comment (lmsurpre) [reply fhir-core/src/main/java/com/ibm/watson/health/fhir/core/util/LRUCache.java]: looks good

**Pull request merged**